src/error.rs

```diff
@@ -5,7 +5,16 @@ use std::{error, fmt, io};
 
 pub use frame::Reason;
 
-/// The error type for HTTP/2 operations
+/// Represents HTTP/2.0 operation errors.
+///
+/// `Error` covers error cases raised by protocol errors caused by the
+/// peer, I/O (transport) errors, and errors caused by the user of the library.
```

Comment:
TIOLI, but maybe "protocol errors caused by the peer", "I/O (transport) errors", and "errors caused by the user of the library" should mention and/or link to the types that represent those individual error cases?

Reply:
All of these cases are not differentiated at the user API level. The error cause is pretty opaque as of now.